### DISCUSSION TITLE
Allowing Socks5 as upstream Proxy

### DISCUSSION THREAD

**Discussion by blird** [mitmproxy/mitmproxy#7159] [Ideas]
I just recently monkey-patched mitm to work with a http proxy that requires authentication for upstream. However I would love to move away from using http(s) and use socks proxies.

**Comment by Prinzhorn** [comment]
> I just recently monkey-patched mitm to work with a http proxy that requires authentication for upstream

To clarify: `--upstream-auth` did not work in your case?

Also for feature requests I recommend opening a feature request (not a discussion) and following the template.

https://github.com/mitmproxy/mitmproxy/issues/new?assignees=&labels=kind%2Ffeature&projects=&template=feature_request.md&title=
